File: android/app/src/main/kotlin/com/swastricare/health/ui/components/OfflineBanner.kt
package com.swastricare.health.ui.components

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutVertically
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.WifiOff
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.swastricare.health.data.services.NetworkMonitorService

@Composable
fun OfflineBanner(networkMonitor: NetworkMonitorService) {
    val isConnected by networkMonitor.isConnected.collectAsState()
    val isSyncing by networkMonitor.isSyncing.collectAsState()

    AnimatedVisibility(
        visible = !isConnected || isSyncing,
        enter = slideInVertically(initialOffsetY = { -it }),
        exit = slideOutVertically(targetOffsetY = { -it })
    ) {
        val bgColor = if (!isConnected) Color(0xFFF59E0B) else Color(0xFF4F46E5)
        val text = if (!isConnected) "You're offline. Some features may be limited." else "Syncing your data..."

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(bgColor)
                .padding(horizontal = 16.dp, vertical = 10.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.Center
        ) {
            if (!isConnected) {
                Icon(
                    imageVector = Icons.Default.WifiOff,
                    contentDescription = null,
                    tint = Color.White,
                    modifier = Modifier.size(16.dp)
                )
            } else {
                CircularProgressIndicator(
                    modifier = Modifier.size(14.dp),
                    strokeWidth = 2.dp,
                    color = Color.White
                )
            }
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = text,
                color = Color.White,
                fontSize = 13.sp
            )
        }
    }
}

File: swastricare-mobile-swift/Views/Components/OfflineBanner.swift
import SwiftUI

struct OfflineBanner: View {
    @ObservedObject private var networkMonitor = NetworkMonitorService.shared

    var body: some View {
        if !networkMonitor.isConnected {
            bannerContent(
                icon: "wifi.slash",
                text: "You're offline. Some features may be limited.",
                color: .orange
            )
        } else if networkMonitor.isSyncing {
            bannerContent(
                icon: nil,
                text: "Syncing your data...",
                color: Color(red: 0.31, green: 0.19, blue: 0.57) // purple
            )
        }
    }

    private func bannerContent(icon: String?, text: String, color: Color) -> some View {
        HStack(spacing: 8) {
            if let icon {
                Image(systemName: icon)
                    .font(.subheadline)
            } else {
                ProgressView()
                    .tint(.white)
                    .scaleEffect(0.7)
            }
            Text(text)
                .font(.subheadline)
        }
        .foregroundColor(.white)
        .padding(.horizontal, 16)
        .padding(.vertical, 10)
        .frame(maxWidth: .infinity)
        .background(color)
        .transition(.move(edge: .top).combined(with: .opacity))
        .animation(.easeInOut(duration: 0.3), value: networkMonitor.isConnected)
        .animation(.easeInOut(duration: 0.3), value: networkMonitor.isSyncing)
    }
}
